.github/workflows/icu4c.yml

```diff
@@ -185,6 +185,25 @@ jobs:
           cd $PREFIX/bin;
           LD_LIBRARY_PATH=../lib ./icuinfo
 
+  gcc-warnings-as-errors:
```

Comment:
Is there any advantage in adding yet one more CI job, instead of simply adding the additonal error checking to the existing `gcc-debug-build-and-test` job?